servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -10098,6 +10139,133 @@ public void deleteAssertionCondition(ResourceContext ctx, String domainName, Str
         dbService.executeDeleteAssertionCondition(ctx, domainName, policyName, assertionId, conditionId, auditRef, caller);
     }
 
+    @Override
+    public void putDomainDependency(ResourceContext ctx, String domainName, String auditRef, String service) {
+        final String caller = ctx.getApiName();
+        logPrincipal(ctx);
+
+        if (readOnlyMode.get()) {
+            throw ZMSUtils.requestError(SERVER_READ_ONLY_MESSAGE, caller);
+        }
+
+        validateRequest(ctx.request(), caller);
+
+        validate(domainName, TYPE_DOMAIN_NAME, caller);
+        validate(service, TYPE_SERVICE_NAME, caller);
```

Comment:
follow up on the comment from the rdl - this is an optional attiribute only valid for system admin auth method. otherwise we need to extract the service name from the principal object. as such, we should check for nulls.